### DISCUSSION TITLE
How do I update the plugin??

### DISCUSSION THREAD

**Discussion by craigconstantine** [SebastianMC/obsidian-custom-sort#7] [Q&A] [answered]
obsidian's community plugins > check for updates...
doesn't "find" any updates for this plugin
I'm stuck on 0.5.189

**Comment by SebastianMC** [comment, marked as answer]
Hi,

the plugin is not on the official list of Obsidian Community Plugins visible at https://obsidian.md/plugins, hence the operation _obsidian's community plugins > check for updates_ doesn't take it into account when checking for updates. What is misleading here is that the plugin, when installed manually or via BRAT, becomes visible on the list in _settings > Community plugins > **Installed plugins**_. It looks that Obsidian shows all installed plugins in that list, regardless of how they were installed and if they are part of the _official_ list of community plugins or not.

Until the plugin reaches the list of https://obsidian.md/plugins **I'd recommend to use BRAT method for installation** (https://github.com/SebastianMC/obsidian-custom-sort#installing-the-plugin-using-brat). It is the most convenient method because **BRAT can update the plugins automatically**.

Obviously, manual installation of the newest release of the plugin also works. To do that please follow the procedure described here https://github.com/SebastianMC/obsidian-custom-sort#manually-installing-the-plugin and override ```main.js``` and ```manifest.json``` with their versions downloaded from the chosen release

**Reply by craigconstantine** [reply]
yes, I'm using BRAT. How do I upgrade this to the current version…

<img width="1091" alt="Screen Shot 2022-09-11 at 20 56 28" src="https://user-images.githubusercontent.com/2278992/189557242-3f10f955-a887-4811-862a-a5d501dc55f9.png">

**Reply by SebastianMC** [reply]
Hi,

the BRAT plugin is great and simplifies the installation and update process a lot. Yet, in my (short) experience with BRAT, it turns out that in some cases it can keep a plugin stuck in some older version. Unclear why. If that's the case on your side, please uninstall the ```obsidian-custom-sort```  in BRAT (the screenshot below) and install it again via BRAT

<img width="1094" alt="Uninstall in BRAT" src="https://user-images.githubusercontent.com/23032356/189596667-460db22f-4d4a-4a24-9213-efd797e6a679.png">

If reinstallation doesn't do the update (it should), use one of BRAT commands to check for available updates:

<img width="829" alt="BRAT command" src="https://user-images.githubusercontent.com/23032356/189600206-8fc5279f-509b-4204-8498-3061c4fff79f.png">

If that doesn't work, the fallback solution seems to be the manual removal of the plugin files after uninstalling it in BRAT. Then, reinstallation via BRAT will by definition download the newest release. In that case keep in mind, that any settings of obsidian-custom-sort will be lost and you will have to re-configure. Looking at your screenshot, the 'any settings' in your case would be the path to custom sorting specification file (which defaults to ```Inbox/Inbox.md```)

As I wrote in the beginning: BRAT is great, yet sometimes it keeps a plugin stuck. To save time, instead of debugging why a plugin got stuck, the fastest path is to reinstall or uninstall, remove and install from scratch (as described above). From the perspective of beta-plugin developer there is not much to be done to support BRAT, the BRAT mechanics works on top of the standard github release process.

**Comment by craigconstantine** [comment]
`  ¯\_(ツ)_/¯ ` I managed to remove and reinstall via BRAT.

**Comment by SebastianMC** [comment]
Probably it makes sense to post a final update here: few weeks ago the plugin reached the official library of Obsidian community plugins. Thus, it can be installed and updated directly from inside the application
